xcarchive/ios.go

```diff
@@ -378,3 +379,38 @@ func (archive IosArchive) BundleIDProfileInfoMap() map[string]profileutil.Provis
 func (archive IosArchive) FindDSYMs() ([]string, []string, error) {
 	return findDSYMs(archive.Path)
 }
+
+// ReadCodesignParameters ...
+func (archive IosArchive) ReadCodesignParameters() (*autocodesign.AppLayout, error) {
+	var teamID string
+	entitlementsMap := map[string]autocodesign.Entitlements{}
+
+	bundleIDProfileInfoMap := archive.BundleIDProfileInfoMap()
+	for bundleIdentifier, provisioningProfile := range bundleIDProfileInfoMap {
+		if teamID == "" {
+			teamID = provisioningProfile.TeamID
+		}
+
+		entitlementsMap[bundleIdentifier] = autocodesign.Entitlements(provisioningProfile.Entitlements)
```

Comment:
the Provisioning Profile's entitlements contains not just codesign-related entitlements (like `get-task-allow`), we should either filter them here or later when using the entitlement.

Reply:
Good point. Had a look at the other usage of the `AppLayout` struct and (if I read it correctly) it returns only the user specified entitlements from the actual project itself. Based on that I would replicate the same behaviour here and only return the codesign related ones.

Reply:
discussed offline that we will leave it as it is currently.